opencog/benchmark/diary.txt

```diff
@@ -1087,7 +1101,39 @@ Again, ProtoAtom.
 AtomPtr column: copied from above
 Atom* column:   using bool atoms_less(const Atom*, const Atom*); in
                 Handle.h (used to be AtomPtr)
-ProtoAtom*:     using bool atoms_less(const ProtAtom*, etc...)
+ProtoAtom*:     using bool atoms_less(const ProtoAtom*, etc...)
 
 Conclude: switching to * from Ptr for handle compares helps. But AddLink
    still hurts; it hurts just a bit less.
+
+======================================================================
+
+4 Feb 2016
+----------
+ProtoAtom work .. how much does a cast really cost?
+i.e. a cast from Handle to NodePtr or LinkPtr ?
+
+atomspace_bm -m noop -m getType -m pointerCast -m getOutgoingSet
+
+                  previous      NodeCast        LinkCast
+   name           K-ops/sec     K-ops/sec      K-ops/sec
+   ----           ---------     ---------      ---------
+  noop              1e6         889K             864K
+  getType          75.5K         72.0K            72.0K
+  pointerCast        -            6375             7354
+  getOutgoingSet    7555          7646             7797
+
+previous: copied column from above.
+
+NodeCast: perform NodeCast only.
+
+LinkCast: perform LinkCast only.
+
+Comments: we expect the pointerCast and the getOutgoingSet values to be
+nearly identical, since getOutgoingSet is just a cast, followed by a
+very simple method call.  The NodeCast and Link Cast numbers might
+differ, but not much, due to the different ratios of links and nodes in
+the sample set.
+
+Ergo, the cost of getOutgoingSet is dominated by the LinkCast!  If we
+get rid of it, then what happens?  It goes up to 44.1M-ops/sec! Woww!
```

Comment:
But benchmarks are needed.  Pattern matcher performance is not really understood.